numpy/testing/_private/utils.py

```diff
@@ -784,18 +785,18 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
         if isinstance(val, bool):
             cond = val
-            reduced = [0]
+            reduced = array([0])
```

Comment:
I think this should be `False` to get the most out of `logical_not` below, and for consistency with the dtype of the `else1` branch?

Reply:
I think it should be `array([cond])` or even `array(cond)` (cond or val same thing), but I didn't want to change the "style too much"

Reply:
Actually, this whole `if` can be simplified to
```python
reduced = np.ravel(val)
cond = reduced.all()
```

Fine to leave as a follow-up - especially if you drop the `logical_not` below in favor of a subtraction

Reply:
We get bit a few `__array_function__` errors here and there
```python
TypeError: no implementation found for 'numpy.ravel' on types that implement __array_function__: [<class 'numpy.core.tests.test_overrides.TestGetImplementingArgs.test_ndarray_subclasses.<locals>.OverrideSub'>]
```